### ISSUE TITLE
Wrong nodejs import path when generating declaration 

### ISSUE THREAD

**Issue by dbellavista** [microsoft/TypeScript#42137]
# Bug Report

### 🔎 Search Terms

- `File '@types/node/events.d.ts' is not a module.`
- import from "node/events"
- require("node/events")

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because versions before 4.0 are unable to generate the correct type definition

### ⏯ Repo Link and Code

Sorry I could't use the playground since I need typings for nodejs.

* [Repo link](https://github.com/dbellavista/tsc-nodeimport-issue)

```bash
npm install
npx tsc -p jsconfig.json
npx tsc -p noemit.jsconfig.json
```

### 🙁 Actual behavior

The generated definitions is 

```ts
export class Issue extends EventEmitter {
}
import { Writable } from "node/stream";
import EventEmitter = require("node/events");
export { Writable };
```

which causes the errors:

```
issue.d.ts:3:26 - error TS2306: File '/home/me/issue/node_modules/@types/node/stream.d.ts' is not a module.

3 import { Writable } from "node/stream";
                           ~~~~~~~~~~~~~

issue.d.ts:4:31 - error TS2306: File '/home/me/issue/node_modules/@types/node/events.d.ts' is not a module.

4 import EventEmitter = require("node/events");

```

### 🙂 Expected behavior

The correct definition file should be

```ts
export class Issue extends EventEmitter {
}
import { Writable } from "stream";
import EventEmitter = require("events");
export { Writable };
```

**Comment by andrewbranch** [comment]
Playground declaration emit actually shows

```ts
export class Issue {
}
import { Writable } from "%40types/node/stream";
export { Writable };
```

👀

**Comment by andrewbranch** [comment]
But seems to be fixed in the nightly? @dbellavista, could you try with `typescript@next`?

**Comment by dbellavista** [comment]
Sorry I haven't notice the declaration on playground and I haven't tested on `@next`.

@andrewbranch it works correctly with `typescript@next`

**Comment by andrewbranch** [comment]
I guess it would be good to find out whether this was fixed on purpose or on accident, and if it’s worth backporting.

**Comment by sheetalkamat** [comment]
This is fixed by #41816 i think,
